controllers/placementapi_controller.go

```diff
@@ -1181,7 +1183,9 @@ func (r *PlacementAPIReconciler) ensureDeployment(
 		return ctrl.Result{}, err
 	}
 
-	if instance.Status.ReadyCount > 0 || *instance.Spec.Replicas == 0 {
+	// We need to check generation with a similar pattern to k8s,
+	// just in case when the status is recreated or the etcd database needs to be recreated
+	if (instance.Status.ReadyCount == *instance.Spec.Replicas && deployment.Generation == deployment.Status.ObservedGeneration) || *instance.Spec.Replicas == 0 {
```

Comment:
Do we need a separate `|| *instance.Spec.Replicas == 0` ? If the Spec.Replicas is 0 then I expect that the ReadyCount will be 0 eventually as well

Reply:
If you remove the `|| *instance.Spec.Replicas == 0` you might want to always initialize ReadyCount, that could be an omitted field as per [1]. I agree it can be removed, provided you set `default=0` and remove `omitempty`, right?

[1] https://github.com/openstack-k8s-operators/placement-operator/blob/main/api/v1beta1/placementapi_types.go#L141

Reply:
I think this logic is independent from the fact if we send back ReadyCount in the Status response or omitting it if the value is 0.

I'm not against removing omitempty from ReadyCount I just not see how that effects this logic when we decide the Deployment has the requested amount of replicas

Reply:
I think in my mind I was looking at `if instance.Status.ReadyCount > 0 || *instance.Spec.Replicas == 0 {`,  but I agree if we compare it w/ the `.Spec.Replicas` we are not going to hit any error

Reply:
agree that when we drop this will still work but IMO there is no reasone for waiting X reconcile cycles to get same result if we know that there is no more things to do

Reply:
If the goal of the client (openstack-operator) is to stop all placementAPIs before execute some major upgrade step, then the placement-operator should not report readyness before the pods are actually removed.